### PR TITLE
Add tech polyfill

### PR DESCRIPTION
**Pull request by misteroneill**
This does a 99% of the way there polyfill of the old tech behavior, exposing methods and properties of the `tech_` object on the `tech` function.

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -157,4 +159,40 @@
...
+  // a mostly-complete implementation of the old tech behavior.
+  Player.prototype.loadTech_ = function() {
+    var player = this;
+
+    originals.Player.loadTech_.apply(player, arguments);
+
+    var tech = this.tech_;
+    var proto = tech.constructor.prototype;
+
+    this.polyfilledTechKeys_ = Object.keys(proto);
```
Comment (gkatsev) [inline src/plugin.js]: why not loop through the keys that are directly on the `tech`? It's theoretically possible there are stuff there that aren't on the prototype.

Comment (misteroneill) [inline src/plugin.js]: Good point.

File: src/plugin.js [hunk 2]
```diff
@@ -157,4 +159,48 @@
...
+
+    var tech = this.tech_;
+    var ownKeys = Object.keys(tech);
+    var proto = tech.constructor.prototype;
+    var protoKeys = Object.keys(proto);
+
+    this.polyfilledTechKeys_ = ownKeys.concat(protoKeys.filter(function(key) {
+
+      // Only take protoKeys that are NOT found in the ownKeys.
+      return ownKeys.indexOf(key) === -1;
```
Comment (dmlap) [inline src/plugin.js:177]: This isn't too bad or anything but I probably would have gone with `tech[key]` for the check instead of the array-search.

Comment (misteroneill) [inline src/plugin.js:177]: How would that work? `tech[key]` would match properties both on the tech itself and up the prototype chain. This way it's explicit.

Comment (dmlap) [inline src/plugin.js:177]: :+1:

**Comment by dmlap** [comment]
LGTM

File: src/plugin.js (hunk 2 above)
Comment (gkatsev) [inline src/plugin.js:177]: tech.hasOwnProperty(key)

**Pull request merged**